var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -68,6 +70,10 @@ class Dd4hep(CMakePackage):
     depends_on('lcio', when="+lcio")
     depends_on('edm4hep', when="+edm4hep")
 
+    # See https://github.com/AIDASoft/DD4hep/pull/771
+    conflicts('^cmake@3.16:3.17.0', when='@1.15:1.16.1',
+              msg='cmake version with buggy FindPython breaks dd4hep cmake config')
```

Comment:
Wouldn't it be easier to say:
```
depends_on('cmake@3.17.1:', when='@1.15:1.16.1')
```
?

Reply:
That would make it a "hard" dependency, right? However, dd4hep works with `cmake@3.12:3.16.2` and then again with `cmake@3.17.1:` for all versions. For me this was easier to express with a `conflict`, but if it is possible to do the same with a dependency, I am all for it.

Reply:
I checked the dd4hep release notes again, and the conflict really only exists for `dd4hep@1.15`, so I think leaving it as a conflict is OK. (Still open for making it a dependency if that is possible).